### PR TITLE
Search fixes, lodash 4.1.0 and the ability to search ideas by comment authors

### PR DESCRIPTION
**Pull request by YashdalfTheGray**
Approvals requested

- [x] @alanlgirard 
- [x] @Mctalian

### CODE REVIEW

**Comment by YashdalfTheGray** [comment]
Travis is full of shit here, I don't know why though. 

Proof (on the `search-fixes` branch): 
screenshot 2016-02-01 23 38 21

Proof 2 (on `master`):
screenshot 2016-02-01 23 39 34

File: server/ideas/ideaModel.js [hunk 1]
```diff
@@ -18,7 +19,7 @@ function Idea(title, description, authorId, eventId, tags, rolesreq) {
     this.comments = [];
     this.backs = [{
         text: "Idea Owner",
-        authorId: this.authorId,
+        authorId: new ObjectId(this.authorId),
```
Comment (Mctalian) [inline server/ideas/ideaModel.js]: Do you need this if you are wrapping authorId in new ObjectId in the routes?

Comment (YashdalfTheGray) [inline server/ideas/ideaModel.js]: You're right, it's not necessary.

**Comment by Mctalian** [comment]
Approved!

**Comment by alanlgirard** [comment]
Approved

**Pull request merged**